src/control/server/mgmt_svc.go

```diff
@@ -125,30 +126,41 @@ func (svc *mgmtSvc) GetAttachInfo(ctx context.Context, req *mgmtpb.GetAttachInfo
 		return nil, errors.New("clientNetworkCfg is missing")
 	}
 
-	var groupMap *system.GroupMap
-	var err error
-	if req.GetAllRanks() {
-		groupMap, err = svc.sysdb.GroupMap()
-	} else {
-		groupMap, err = svc.sysdb.ReplicaRanks()
-	}
+	groupMap, replicaRanks, err := svc.sysdb.GroupMapWithReplicaRanks()
 	if err != nil {
 		return nil, err
 	}
 
 	resp := new(mgmtpb.GetAttachInfoResp)
-	for rank, uri := range groupMap.RankURIs {
-		resp.Psrs = append(resp.Psrs, &mgmtpb.GetAttachInfoResp_Psr{
-			Rank: rank.Uint32(),
-			Uri:  uri,
-		})
+	if req.GetAllRanks() {
+		for rank, uri := range groupMap.RankURIs {
+			resp.RankUris = append(resp.RankUris, &mgmtpb.GetAttachInfoResp_RankUri{
+				Rank: rank.Uint32(),
+				Uri:  uri,
+			})
+		}
 	}
 	resp.Provider = svc.clientNetworkCfg.Provider
 	resp.CrtCtxShareAddr = svc.clientNetworkCfg.CrtCtxShareAddr
 	resp.CrtTimeout = svc.clientNetworkCfg.CrtTimeout
 	resp.NetDevClass = svc.clientNetworkCfg.NetDevClass
+	for _, rank := range replicaRanks {
+		resp.MsRanks = append(resp.MsRanks, rank.Uint32())
+	}
+
+	// For resp.RankUris may be large, we make a resp copy with a limited
```

Comment:
I agree that it's a good idea to limit this to one message to avoid log spam.  That said, it might be worth getting some opinions if you should use one of the debug options (or a new one) to increase the message count to include all or just more, of the data.  It seems like it might be a good idea to have access to that data in the near future without having to modify the code to get it.

Reply:
Somehow I thought I was reading this in the agent, but now I see that it is not.  The agent supports some extended debug options such as "net", which we use to provide additional debug info when desired.  I don't expect @mjmac will want to add more debug options to the server, but if he does, then there's a possibility there.

Reply:
I see. We do log all the other fields in resp completely, by the way, just in case there's misunderstanding there. I'll leave the limit of resp.RankUris as it is then. Future improvements are surely welcome.

Reply:
Given how important this response is to the functioning of the system, it might be worth adding a custom debug printer for the response here. It adds a bit more maintenance overhead to keep it updated if the response fields change, but I think it's worthwhile.

In particular, I think it would be useful to dump the fields in a condensed format similar to how we do it on the client side. I also think that it would be good to see the total length of RankUris in addition to printing a map of MsRanks -> URIs. I would expect that this should be a reasonable length even on large systems. My thinking here is that it may be most helpful to see any problems with the URIs associated with MS ranks rather than arbitrarily printing a single mapping.

Reply:
len(RankUris) is already printed toward the end. I'll see what I can do...

Reply:
@mjmac, could the control plane team help implement such a change after this patch lands? I'm not happy with the resulting code of my own attempt.

Reply:
Sure, that's fine.